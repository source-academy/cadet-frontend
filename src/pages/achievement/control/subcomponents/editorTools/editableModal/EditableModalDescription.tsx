import { EditableText } from '@blueprintjs/core';
import React from 'react';

type EditableModalDescriptionProps = {
  description: string;
  setDescription: any;
};

function EditableModalDescription(props: EditableModalDescriptionProps) {
  const { description, setDescription } = props;

  return (
    <>
      <h3>
        <EditableText
          placeholder={`Enter your description here`}
          value={description}
          onChange={setDescription}
          multiline={true}
        />
      </h3>
    </>
  );
}

export default EditableModalDescription;
