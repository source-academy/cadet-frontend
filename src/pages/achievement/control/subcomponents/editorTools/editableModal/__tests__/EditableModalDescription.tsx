import { mount } from 'enzyme';
import React from 'react';

import EditableModalDescription from '../EditableModalDescription';

const mockProps = {
  description: 'Sample Description',
  setDescription: () => {}
};

test('EditableModalDescription component renders correctly', () => {
  const description = <EditableModalDescription {...mockProps} />;
  const tree = mount(description);
  expect(tree.debug()).toMatchSnapshot();
});
