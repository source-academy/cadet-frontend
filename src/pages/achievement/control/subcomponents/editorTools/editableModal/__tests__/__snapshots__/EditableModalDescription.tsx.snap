// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableModalDescription component renders correctly 1`] = `
"<EditableModalDescription description=\\"Sample Description\\" setDescription={[Function: setDescription]}>
  <h3>
    <Blueprint3.EditableText placeholder=\\"Enter your description here\\" value=\\"Sample Description\\" onChange={[Function: setDescription]} multiline={true} alwaysRenderInput={false} confirmOnEnterKey={false} defaultValue=\\"\\" disabled={false} maxLines={Infinity} minLines={1} minWidth={80} type=\\"text\\">
      <div className=\\"bp3-editable-text bp3-multiline\\" onFocus={[Function (anonymous)]} tabIndex={0}>
        <span className=\\"bp3-editable-text-content\\" style={{...}}>
          Sample Description
        </span>
      </div>
    </Blueprint3.EditableText>
  </h3>
</EditableModalDescription>"
`;
